apps/src/p5lab/layoutUtils.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * Given a group of sprites, arrange all sprites of this type in a particular
+ * layout. This is likely to change some or all of position/rotation/scale for
+ * the sprites in the group.
+ * Entirely derived from the logic for similar blocks in our dance party repo here (more layouts supported there):
+ * https://github.com/code-dot-org/dance-party/blob/6672bdb0cffad1cbfda6e7396155f542b6cdcffe/src/p5.dance.js#L637-L911
+ */
+export function layoutSpriteGroup(group, layout, p5) {
+  const count = group.length;
+  if (!count) {
+    return;
+  }
+
+  const minX = 20;
+  const maxX = 400 - minX;
+  const minY = 35;
+  const maxY = 400 - 40;
+  const radiansToDegrees = 180 / Math.PI;
+  const maxCircleRadius = 165;
+
+  if (layout === 'top') {
+    createRow(group, 100);
+  } else if (layout === 'row') {
+    createRow(group, 200);
+  } else if (layout === 'bottom') {
+    createRow(group, 300);
+  } else if (layout === 'border') {
```

Comment:
Nit: Would using a `switch` to check the layout type improve readability?  Some of the specific layout types have their own if/else-if/else logic already.

Reply:
I decided against ES6-ifying/cleaning up all of this for a couple reasons, LMK if you strongly disagree:
- if future improvements are needed across both dance and spritelab, will require developers to have to translate between ES6 and older syntax -- leaving as-is makes making changes in the same places easier to identify/make. If we want to refactor, we should refactor both
- feels out of scope for this work to just add this functionality in spritelab to also refactor it

Reply:
I think reusing a the dance code and making a common layouts file could help a lot here. One thing about how this is written that makes me nervous is the large chunks of code between the main if/else statements. Whether it moves to a switch or not, it would be good to move the logic into separate functions so the layoutSpriteGroup function is only handling which function to call. (i.e. I'm thinking about the case where a future maintainer misplaces a curly brace and ends up functionally removing a bunch of layouts accidentally)

Reply:
Moved core logic to helper functions, think that's an improvement even if we don't do full dance party refactor right now. LMK what you think!